src/main/resources/db/migrations/V001__CreateSchemaTables.sql

```diff
@@ -0,0 +1,171 @@
+    room_id UUID REFERENCES rooms(id) NOT NULL
+);
+
+CREATE TABLE public.cases (
+    id UUID PRIMARY KEY NOT NULL,
+    court_id UUID REFERENCES courts(id) NOT NULL ,
+    case_ref VARCHAR(25) NOT NULL, 
+    test BOOLEAN DEFAULT FALSE,
+    deleted BOOLEAN DEFAULT FALSE,
+    created_on TIMESTAMPTZ NOT NULL DEFAULT NOW(),
+    modified_on TIMESTAMPTZ DEFAULT NOW()
+);
+
+
+CREATE TABLE public.participants (
+    id UUID PRIMARY KEY NOT NULL,
+    case_id UUID REFERENCES cases(id) NOT NULL,
+    participant_type PARTICIPANT_TYPE NOT NULL,
+    first_name VARCHAR(100) NOT NULL,
+    last_name VARCHAR(100) NOT NULL,
+    deleted BOOLEAN DEFAULT FALSE,
+    created_on TIMESTAMPTZ NOT NULL DEFAULT NOW(),
+    modified_on TIMESTAMPTZ DEFAULT NOW()
+);
+
+CREATE TABLE public.bookings (
+    id UUID PRIMARY KEY NOT NULL,
+    case_id UUID REFERENCES cases(id) NOT NULL,
+    date DATE NOT NULL,
+    deleted BOOLEAN DEFAULT FALSE,
+    created_on TIMESTAMPTZ NOT NULL DEFAULT NOW(),
+    modified_on TIMESTAMPTZ DEFAULT NOW()
+);
+
+
+CREATE TABLE public.booking_participants (
+    id UUID PRIMARY KEY NOT NULL,
+    participant_id UUID REFERENCES participants(id) NOT NULL,
+    booking_id UUID REFERENCES bookings(id) NOT NULL
+);
+
+CREATE TABLE public.users (
+    id UUID PRIMARY KEY NOT NULL,
+    first_name VARCHAR(100) NOT NULL,
+    last_name VARCHAR(100) NOT NULL,
+    email VARCHAR(100) NOT NULL,
+    organisation VARCHAR(250),
+    phone VARCHAR(50),
+    deleted BOOLEAN DEFAULT FALSE,
+    created_on TIMESTAMPTZ DEFAULT NOW(),
+    modified_on TIMESTAMPTZ DEFAULT NOW()
+);
+
+CREATE TABLE public.roles (
+    id UUID PRIMARY KEY NOT NULL,
+    name VARCHAR(45) NOT NULL
+);
+
+CREATE TABLE public.app_access (
+    id UUID PRIMARY KEY NOT NULL,
+    user_id UUID REFERENCES users(id) NOT NULL,
+    court_id UUID REFERENCES courts(id) NOT NULL,
+    role_id UUID REFERENCES roles(id) NOT NULL,
+    last_access TIMESTAMPTZ,
+    active BOOLEAN DEFAULT TRUE,
+    deleted BOOLEAN DEFAULT FALSE,
+    created_on TIMESTAMPTZ DEFAULT NOW(),
+    modified_on TIMESTAMPTZ DEFAULT NOW()
+);
+
+
+CREATE TABLE public.capture_sessions (
+    id UUID PRIMARY KEY NOT NULL,
+    booking_id UUID REFERENCES bookings(id) NOT NULL,
+    origin RECORDING_ORIGIN NOT NULL,
+    ingest_address VARCHAR(255),
+    live_output_url VARCHAR(100),
+    started_on TIMESTAMPTZ,
+    started_by_user_id UUID REFERENCES users(id),
+    finished_on TIMESTAMPTZ,
+    finished_by_user_id UUID REFERENCES users(id),
+    status RECORDING_STATUS,
+    deleted BOOLEAN DEFAULT FALSE
+);
+
+CREATE TABLE public.recording_versions (
+    id UUID PRIMARY KEY NOT NULL,
+    capture_session_id UUID REFERENCES capture_sessions(id) NOT NULL,
+    version INT NOT NULL,
+    url VARCHAR(255) NOT NULL,
+    filename VARCHAR(255) NOT NULL,
+    created_on TIMESTAMPTZ NOT NULL,
+    duration TIME,
+    edit_instruction JSON,
+    deleted BOOLEAN DEFAULT FALSE
+);
+
+CREATE TABLE public.share_recordings (
```

Comment:
i think there is a way that we can make sharing resources easier within the system. i propose we created a bi-directional polymorphic invitations table where any model can be attached to any other model, this would allow:
- courts to share recordings to external users
- courts to share a recording with another court
- courts to share a whole case with another court
- etc etc...

the columns needed for this would be:
- `id`
- `invitable_type` -> e.g. recording
- `invitable_id`
- `invited_type` -> e.g. user
- `invited_id`
- `permissions` -> create/read/update/delete

who created the invite would be captured by the audit table.

Reply:
We will revisit in the future